### ISSUE TITLE
Identity property

### ISSUE THREAD

**Issue by PushOrigin** [MarimerLLC/cslaforum#520]
I updated a project from 4.6.5 to 4.7.1 and had to implement an IBusinessObject.Identity property on all of my IReadOnlyObject based classes. Due to an ill-advised database design decision made a long time ago (by me unfortunately) some of those classes have a Guid primary key.

How should classes that don't have an int identity implement the Identity property? At this point I'm using a hash over two or three properties. Just curious how it should be handled, it seems like there are other more legitimate cases where the identity might not be an int.

**Comment by rockfordlhotka** [comment]
That new `Identity` property relates to a serialization optimization feature, and _must_ be implemented for that purpose only. **This property has no business or database purpose!** This comes out of https://github.com/MarimerLLC/csla/issues/712.

You can see how the property _must_ be implemented by looking at the existing CSLA base classes, such as `BusinessBase`: 
https://github.com/MarimerLLC/csla/blob/b51761b54350f99a9c07fa73e28dedb494a10be3/Source/Csla.Shared/Core/BusinessBase.cs#L86.

The `Identity` property must return a value that is unique within the current object graph. This value is used to merge the result of `SaveAsync` back into the existing graph (if desired). And to make that happen, each object in the graph must have a unique id value.

**Comment by PushOrigin** [comment]
Aah, got it. At some point I started IReadOnlyObject instead of
ReadOnlyBase<T>, not sure why. I see that ReadOnlyBase<T> just returns 0
for Identity. I'm replacing IReadOnlyObject with ReadOnlyBase<T>. Nothing
like causing my own problems...

**Comment by rockfordlhotka** [comment]
Yes, the read-only types just return 0 because you can't save them - so there's no scenario under which you should need to merge the result of `SaveAsync` back into a read-only object graph.
